### DISCUSSION TITLE
Is it possible to change the subplot of 'sigma' in plot_fit_delchi() to the histogram style?

### DISCUSSION THREAD

**Discussion by judychang2023** [sherpa/sherpa#1831] [Q&A]
Hi All,
The 'sigma' y-axis of the delchi plot is definded by the (data - model) divided by the error, where the error in this case are the statistical uncertainties that are typically calculated during the model fitting.

It always returns to a value "1" from the returns of get_fit_delchi().

Since yerror has nothing to do with the error bars in the delchi plot, I was just wondering if plot_fit_delchi could be styled as a histogram without the need of showing the yerror?

Thank you for kindly help~

Best Regards,
Judy

**Comment by DougBurke** [comment]
Judy, this doesn't quite answer your question, but you can change the look of the resid/sigma/delchi plots - e.g. see https://github.com/sherpa/sherpa/issues/1817#issuecomment-1658504691

**Reply by DougBurke** [reply]
So, using the 4.16 release, you can say

```
prefs = get_plot_prefs('delchi')
prefs['linestyle'] = 'solid'
prefs['yerrorbars'] = False
plot_delchi()
```

but it doesn't draw a histogram-style plot. Boring technical talk: we would need to add `Data1DInt/DataPHA` variants of `sherpa.plot.DelchiPlot` (and probably the other residual-style plots) as we did in #906 (or perhaps, this could have been included in #906)

**Reply by DougBurke** [reply]
The 4.17.0 Sherpa release (or the CIAO 4.17 release in mid December) has changed how the residual-style plots are displayed for PHA data, so it should look more like a histogram. You may still need to turn off the y error bars, which can be done with something like

```
get_plot_prefs('delchi')['yerrorbars'] = False
```

**Reply by DougBurke** [reply]
For reference, with CIAO 4.17 we can say (for a PHA dataset)

```python
plot("delchi", "delchi", yerrorbars=[True, False])
```

and get

![sherpa](https://github.com/user-attachments/assets/d7e7505e-e60d-4b96-9150-c040443d7319)
